modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/UserCardTag.java

```diff
@@ -0,0 +1,512 @@
+			jspWriter.write("class=\"aspect-ratio-item-fluid\" src=\"");
+			jspWriter.write(_imageSrc);
+			jspWriter.write("\"	/>");
+		}
+		else {
+			jspWriter.write("<svg class=\"lexicon-icon lexicon-icon-user\" ");
+			jspWriter.write("role=\"presentation\" viewBox=\"0 0 512 512\">");
+			jspWriter.write("<use xlink:href=\"");
+			jspWriter.write(_spritemap);
+			jspWriter.write("#user\" /></svg>");
+		}
+
+		jspWriter.write("</span>");
+		jspWriter.write("</span>");
+		jspWriter.write("</div>");
+
+		if (_selectable) {
+			jspWriter.write("</label>");
+			jspWriter.write("</div>");
+		}
+
+		jspWriter.write("</div>");
+
+		jspWriter.write("<div class=\"card-body\">");
+		jspWriter.write("<div class=\"card-row\">");
+		jspWriter.write("<div class=\"autofit-col autofit-col-expand\">");
+		jspWriter.write("<p class=\"card-title\">");
+		jspWriter.write("<span class=\"text-truncate-inline\">");
+
+		if (!_disabled && Validator.isNotNull(_href)) {
+			jspWriter.write("<a class=\"text-truncate\" href=\"");
+			jspWriter.write(_href);
+			jspWriter.write("\">");
+			jspWriter.write(_name);
+			jspWriter.write("</a>");
+		}
+		else {
+			jspWriter.write("<span class=\"text-truncate\">");
+			jspWriter.write(_name);
+			jspWriter.write("</span>");
+		}
+
+		jspWriter.write("</span>");
+		jspWriter.write("</p>");
+		jspWriter.write("<p class=\"card-subtitle\">");
+		jspWriter.write("<span class=\"text-truncate-inline\">");
+		jspWriter.write("<span class=\"text-truncate\">");
+		jspWriter.write(_subtitle);
+		jspWriter.write("</span>");
+		jspWriter.write("</span>");
+		jspWriter.write("</p>");
+
+		if (!ListUtil.isEmpty(_labels)) {
+			jspWriter.write("<div class=\"card-detail\">");
+
+			for (LabelItem labelItem : _labels) {
+				jspWriter.write("<span class=\"label ");
+
+				if ((boolean)labelItem.get("dismissible")) {
+					jspWriter.write("label-dismissible ");
+				}
+
+				if ((boolean)labelItem.get("large")) {
+					jspWriter.write("label-lg ");
+				}
+
+				if (Validator.isNotNull(labelItem.get("displayType"))) {
+					jspWriter.write(
+						"label-" + labelItem.get("displayType") + " ");
+				}
+
+				jspWriter.write("\">");
+				jspWriter.write(String.valueOf(labelItem.get("label")));
+				jspWriter.write("</span>");
+			}
+
+			jspWriter.write("</div>");
+		}
+
+		jspWriter.write("</div>");
+
+		if (!ListUtil.isEmpty(_actionDropdownItems)) {
+			jspWriter.write("<div class=\"dropdown\">");
+			jspWriter.write("<div class=\"dropdown-toggle component-action\">");
+			jspWriter.write("<svg class=\"lexicon-icon lexicon-icon-user\" ");
+			jspWriter.write("role=\"presentation\" viewBox=\"0 0 512 512\">");
+			jspWriter.write("<use xlink:href=\"");
+			jspWriter.write(_spritemap);
+			jspWriter.write("#ellipsis-v\" /></svg>");
+			jspWriter.write("</div>");
+			jspWriter.write("</div>");
+		}
+
+		jspWriter.write("</div>");
+		jspWriter.write("</div>");
+		jspWriter.write("</div>");
+
+		return SKIP_BODY;
```

Comment:
`#n00b alert`

I'm just wondering why we return `SKIP_BODY`;
Probably off topic to ask this here, but I'm interested if you have an explanation.

Reply:
https://lmgtfy.app/?q=jsp+tag+skip_body

You can read more about it in any reference for [Tag](https://tomcat.apache.org/tomcat-5.5-doc/jspapi/javax/servlet/jsp/tagext/Tag.html).

I'm not an expert myself, but the return value instructs what to do next with the tag. In case where the tag won't have a `body` or `children` you can skip analyzing it. That's the difference between:

```
<clay:tag attribute="something" />

<clay:tag_with_body>
    This is the body that needs to be evaluated
</clay:tag_with_body>
```

Maybe @izaera can add more to this... and in the future maybe write this down in a guideline document 😂